File: 1701.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        int time = 0;
        double wait = 0;
        int n = customers.size();
        for (int i = 0; i < n; i++) {
            time = max(time, customers[i][0]) + customers[i][1];
            wait += time - customers[i][0];
        }
        return wait / n;
    }
};

int main() {
    Solution sol;
    int n;
    cin >> n;
    vector<vector<int>> customers(n, vector<int>(2));
    for (int i = 0; i < n; i++) {
        cin >> customers[i][0] >> customers[i][1];
    }
    cout << sol.averageWaitingTime(customers) << endl;
    return 0;
}

File: python/1701.py
class Solution:
    def averageWaitingTime(self, customers: list[list[int]]) -> float:
        start = 0
        end = 0
        wait = 0
        for arrive, cost in customers:
            start = max(end, arrive)
            end = start + cost
            wait += end - arrive
        return wait / len(customers)
        
if __name__ == '__main__':
    s = Solution()
    n = int(input())
    customers = []
    for _ in range(n):
        customers.append(list(map(int, input().split())))
    print(s.averageWaitingTime(customers))